### PR TITLE
Insta Aave Automation connector

### PR DESCRIPTION
**Pull request by pradyuman-verma**
- [x] review @thrilok209 
- [x] deploy
- [x] comment
- [x] enable
- [x] add to sdk

### CODE REVIEW

File: contracts/polygon/connectors/aave/automation-v3/main.sol [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		InstaAaveAutomation(0x7eE533CB0642f18191D2927bdA735c3830B355BB);
+
+	function submitAutomationRequest(
+		uint256 safeHealthFactor,
+		uint256 thresholdHealthFactor
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
```
Comment (thrilok209) [inline contracts/polygon/connectors/aave/automation-v3/main.sol:24]: @pradyuman-verma We need to add automation contract as auth.

File: contracts/polygon/connectors/aave/automation-v3/main.sol [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			abi.encode(safeHealthFactor, thresholdHealthFactor)
+		);
+	}
+
+	function cancelAutomationRequest()
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		automation.cancelAutomationRequest();
```
Comment (thrilok209) [inline contracts/polygon/connectors/aave/automation-v3/main.sol:48]: @pradyuman-verma we need to remove automation contract as auth

File: contracts/polygon/connectors/aave/automation-v3/main.sol [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+		(_eventName, _eventParam) = (
+			"LogUpdateAutomation(uint256,uint256)",
+			abi.encode(safeHealthFactor, thresholdHealthFactor)
+		);
+	}
+}
+
+contract ConnectV2InstaAaveAutomation is Resolver {
+	string public constant name = "Insta-Aave-Automation-v1";
```
Comment (thrilok209) [inline contracts/polygon/connectors/aave/automation-v3/main.sol]:
```suggestion
contract ConnectV2InstaAaveV3Automation is Resolver {
	string public constant name = "Insta-Aave-V3-Automation-v1";
```

File: contracts/polygon/connectors/aave/automation-v3/main.sol (hunk 1 above)
Comment (pradyuman-verma) [reply contracts/polygon/connectors/aave/automation-v3/main.sol:24]: Alright

File: contracts/polygon/connectors/aave/automation-v3/main.sol (hunk 2 above)
Comment (pradyuman-verma) [reply contracts/polygon/connectors/aave/automation-v3/main.sol:48]: Okay

**Comment by pradyuman-verma** [comment]
### Deployed on Polygon

Name: `AAVE-V3-AUTOMATION-A`
Addr: [0x8a1c5D4C6Ee7fC9d8959109413a8E67D0402c650](https://polygonscan.com/address/0x8a1c5D4C6Ee7fC9d8959109413a8E67D0402c650#code)
tx: [here](https://polygonscan.com/tx/0x9c622b38ca9d67011ed742cc15ef60af250f985f7a745ac296dbc361d593eb32)


### Deployed on Avax

Name: `AAVE-V3-AUTOMATION-A`
Addr: [0x724B1e5DBc51309b69513A2716c8659051175EB8](https://snowtrace.io/address/0x724B1e5DBc51309b69513A2716c8659051175EB8#code)
tx: [here](https://snowtrace.io/tx/0xc30e936091a1e26656713d9f14777b58bedc7d317e57539cbeed8e49353de436)

**Comment by pradyuman-verma** [comment]
### Deployed on Polygon

Name: `AAVE-V3-AUTOMATION-STAGING-A`
Addr: [0x9F6b9bFCcA258ed9517284F5F087fD9518AFc378](https://polygonscan.com/address/0x9F6b9bFCcA258ed9517284F5F087fD9518AFc378#code)
tx: [here](https://polygonscan.com/tx/0x1e302e9c207027b666095c2e075f82802c6ba33871ec8b0fa89217d940f21eae)


### Deployed on Avax

Name: `AAVE-V3-AUTOMATION-STAGING-A`
Addr: [0xDA431D8BBa93E95DC3C18614dB6e5e03BF8dA09E](https://snowtrace.io/address/0xDA431D8BBa93E95DC3C18614dB6e5e03BF8dA09E#code)
tx: [here](https://snowtrace.io/tx/0x1f0bf861d19da95978f9fe33e5ded3133e3163e4d50e551cb1b02ec278121fe9)

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**